File: Java/Easy/Accepted/circular-sentence_5.java
class Solution {
    public boolean isCircularSentence(String sentence) {
        
        String[] words = sentence.split(" ");
        int n = words.length;
        if (n == 1) {
            return words[0].charAt(0) == words[0].charAt(words[0].length() - 1);
        }
        for (int i = 0; i < n; i++) {
            if (words[i].charAt(words[i].length() - 1) != words[(i + 1) % n].charAt(0)) {
                return false;
            }
        }
        return true;
    }
}

File: Rust/Easy/WrongAnswer/circular-sentence_5.rs
impl Solution {
    pub fn is_circular_sentence(sentence: String) -> bool {
        
        let mut first = sentence.chars().next().unwrap();
        let mut last = sentence.chars().last().unwrap();
        if first.is_uppercase() {
            first = first.to_lowercase().next().unwrap();
        }
        if last.is_uppercase() {
            last = last.to_lowercase().next().unwrap();
        }
        last == first
    }
}